File: articles/astrophysics-and-the-life-of-stars/c/stellar_scaling_table.c
/*
Stellar Scaling Table

This C workflow computes simple main-sequence scaling values:

    L = M^3.5
    t = M / L

It is a low-level educational example, not a calibrated stellar model.
*/

#include <math.h>
#include <stdio.h>

double luminosity_from_mass(double mass_solar) {
    return pow(mass_solar, 3.5);
}

double lifetime_relative(double mass_solar, double luminosity_solar) {
    return mass_solar / luminosity_solar;
}

int main(void) {
    const double masses_solar[] = {0.2, 0.5, 1.0, 2.0, 5.0, 10.0, 20.0, 40.0};
    const int n_masses = 8;

    printf("mass_solar,luminosity_solar_scaling,lifetime_relative_to_sun\n");

    for (int i = 0; i < n_masses; i++) {
        double mass = masses_solar[i];
        double luminosity = luminosity_from_mass(mass);
        double lifetime = lifetime_relative(mass, luminosity);

        printf("%.6f,%.6e,%.6e\n", mass, luminosity, lifetime);
    }

    return 0;
}

File: articles/astrophysics-and-the-life-of-stars/fortran/stellar_scaling_table.f90
! Stellar Scaling Table
!
! This Fortran workflow generates a simple main-sequence scaling table:
!
!   L = M^3.5
!   t = M / L
!
! It is a classic scientific-computing style table generator.

program stellar_scaling_table
  implicit none

  integer :: i
  real(8), dimension(8) :: masses_solar
  real(8) :: mass_solar
  real(8) :: luminosity_solar
  real(8) :: lifetime_relative

  masses_solar = (/0.2d0, 0.5d0, 1.0d0, 2.0d0, 5.0d0, 10.0d0, 20.0d0, 40.0d0/)

  print *, "mass_solar,luminosity_solar_scaling,lifetime_relative_to_sun"

  do i = 1, 8
     mass_solar = masses_solar(i)
     luminosity_solar = mass_solar**3.5d0
     lifetime_relative = mass_solar / luminosity_solar

     print '(F10.4,A,ES14.6,A,ES14.6)', mass_solar, ",", luminosity_solar, ",", lifetime_relative
  end do

end program stellar_scaling_table
